### ISSUE TITLE
Embedded Javascript expressions?

### ISSUE THREAD

**Issue by gersmann** [TroyAlford/react-jsx-parser#43]
Hi Troy, great project. One question, from looking at the code it seems as if react-jsx-parser doesn't support embedded Javascript expressions. Is this correct?

**Comment by TroyAlford** [comment]
@gersmann - that's correct. I have not added any support for that, as of yet. If you give me an example of what you're wanting to do, however, I may be able to give you a good mechanism for it?

**Comment by gersmann** [comment]
@TroyAlford we were looking into doing client side template compilation comparable to Vue templates. We would need to be able to pass props and register handlers to JSX elements within the templates.

**Comment by manuelJung** [comment]
possible workaround: we can manually eval scripts after the component mounts

`
import React from 'react'
import Parser from 'react-jsx-parser'
import Component from './Component'

const html = `
  <h1>Headline</h1>
  <p>Paragraph</p>
  <Component title='Hello World'/>
  <script>alert('Hello World')</script>
`

class App extends React.Component {

  componentDidMount(){
    var extractscript = /<script[\s\S]*<\/script>/gi.exec(html)
    if(!extractscript) return
    var script = extractscript[0].split('>')[1].split('</script')[0]

    // eslint-disable-next-line
    window.eval(script)
  }

  render() {
    return (
      <div className="App">
        <Parser
          components={{Component}}
          jsx={html}
        />
      </div>
    );
  }
}

export default App;

`

**Comment by manuelJung** [comment]
possible workaround: we can manually eval the the script after the component did mount:

```javascript
import React from 'react'
import Parser from 'react-jsx-parser'
import Component from './Component'

const html = `
  <h1>Headline</h1>
  <p>Paragraph</p>
  <Component title='Hello World'/>
  <script>alert('Hello World')</script>
`

class App extends React.Component {

  componentDidMount(){
    var extractscript = /<script[\s\S]*<\/script>/gi.exec(html)
    if(!extractscript) return
    var script = extractscript[0].split('>')[1].split('</script')[0]

    // eslint-disable-next-line
    window.eval(script)
  }

  render() {
    return (
      <div className="App">
        <Parser
          components={{Component}}
          jsx={html}
        />
      </div>
    );
  }
}

export default App;

```

**Comment by gersmann** [comment]
@manuelJung thank you for the suggestion. we decided to go with Vue which brings this capability out of the box. Would have preferred blueprint.js, but element.js is also a feasible option.

**Comment by TroyAlford** [comment]
Yeah :\ At the moment, I don't think this lib has a good, clean way to provide what you're doing. It is definitely a reasonable thing to want, however - and I'd be totally open to a PR for it (or to help with it).

The issue is that in a real webpack scenario, when you write `<MyComponent prop={someFn} />`, `someFn` is available in scope. In this scenario, that's not the case.

I suspect the "right" way to do it, within this lib, would be to resolve another type of `AcornJSX`-parsed symbol, and then attempt to match the name of that symbol up to one of the props passed in `bindings`. In that case, your code would look something like:

```
<JsxParser
  bindings={{ 
    fancyClickHandler() { /* do some stuff */ },
  }}
  jsx="<div onClick={fancyClickHandler} />"
/>
```

The problem, at the moment, is that `react-jsx-parser` does not handle this type of resolution. The best you could do, currently, is:

```
<JsxParser
  bindings={{
    onClick() { /* do some stuff */ },
  }}
  jsx="<div />"
/>
```

Unfortunately, this will apply `onClick` to _every_ child element the parser parses.

**Comment by TroyAlford** [comment]
Actually - I just realized that you're also just trying to get a `<script>` to run. In that case - you will need to override `blacklistedTags` - whose default value is `['script']` - making it strip all `script` tags.

```
<JsxParser
  blacklistedTags={[]}
  components={{Component}}
  jsx={html}
/>
```

This should stop stripping the scripts out of your renders.

**Comment by TroyAlford** [comment]
For anyone interested in this implementation - I've now implemented the above, and it will go out with 1.3.0. This is a *breaking API change* for anyone who was using the `bindings` hash prior.

Specifically, previous versions would include the whole `bindings` object as `props` for every parsed element. In 1.3.0, the `bindings` prop instead becomes a set of named values which can be bound to the components parsed from the JSX.

Example:
```javascript
<JsxParser
  bindings={{
    foo="Foo"
    onClickHandler={(event) => { /* do stuff */ }}
  }}
  blacklistedAttrs={[]} // important, if you're binding on* events
  jsx="<MyComponent foo={foo} onClick={onClickHandler} />"
/>
```
